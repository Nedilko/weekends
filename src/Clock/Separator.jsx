import PropTypes from 'prop-types'

Separator.propTypes = {
  separator: PropTypes.string,
}

function Separator({ separator }) {
  const currentSeparator = separator || ':'
  return <div>{currentSeparator}</div>
}

export default Separator
