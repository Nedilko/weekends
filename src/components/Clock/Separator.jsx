import PropTypes from 'prop-types'

Separator.propTypes = {
  symbol: PropTypes.string,
}

function Separator({ symbol }) {
  return <div className="text-5xl mt-4">{symbol}</div>
}

export default Separator
