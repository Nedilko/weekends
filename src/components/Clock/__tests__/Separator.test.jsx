import Separator from '@components/Clock/Separator'
import { render, screen } from '@utils/test-utils'

describe('Separator', () => {
  it('should render with custom symbol', () => {
    render(<Separator symbol="::" />)
    const separator = screen.getByText(/::/i)
    expect(separator).toBeInTheDocument()
    expect(separator).toMatchSnapshot()
  })
})
