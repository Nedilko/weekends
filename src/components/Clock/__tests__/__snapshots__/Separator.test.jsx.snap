// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator should render with custom symbol 1`] = `
<div
  class="mt-8 flex items-center justify-center pb-2 text-5xl"
>
  ::
</div>
`;

exports[`Separator should render with default symbol 1`] = `
<div
  class="mt-8 flex items-center justify-center pb-2 text-5xl"
>
  :
</div>
`;
